### PR TITLE
Fix units warnings

### PR DESCRIPTION
**Pull request by mhvk**
Further progress towards warnings-free testing! This time, no bugs in the code found, but did find some bugs in the test routines.

### CODE REVIEW

File: astropy/units/tests/test_format.py [hunk 1]
```diff
@@ -429,9 +489,14 @@ def test_vounit_custom():
 
 
 def test_vounit_implicit_custom():
-    x = u.Unit("furlong/week", format="vounit")
+    # Yikes, really...
```
Comment (pllim) [inline astropy/units/tests/test_format.py]: Why the "yikes"? Can you please clarify?

File: astropy/units/tests/test_units.py [hunk 2]
```diff
@@ -160,8 +160,6 @@ def test_unknown_unit():
 
 
 def test_multiple_solidus():
-    assert u.Unit("m/s/kg").to_string() == u.m / u.s / u.kg
```
Comment (pllim) [inline astropy/units/tests/test_units.py]: Is this removal on purpose? It sounds like the multiple calls were needed for a test named `test_multiple_solidus` but I am just guessing.

Comment (mhvk) [reply astropy/units/tests/test_units.py]: It just duplicates the next test, but without a `catch_warnings`

File: astropy/units/tests/test_format.py (hunk 1 above)
Comment (mhvk) [reply astropy/units/tests/test_format.py]: Clarified:
```
    # Yikes, this becomes "femto-urlong"...  But at least there's a warning.
```

**Comment by mhvk** [comment]
@pllim - thanks for the comments. Addressed. I did `skip ci` since the previous run worked (well, apart from a false-positive from coverage)

**Comment by pllim** [comment]
Thanks, @mhvk !
